File: BotfolioChatExampleObjC/BotfolioChatExampleObjC/ViewController.m
//
//  ViewController.m
//  BotfolioChatExampleObjC
//
//  Created by Aleks Mutlu on 13/12/2016.
//  Copyright © 2016 Lena Software. All rights reserved.
//

#import "ViewController.h"
#import <BotfolioChat/BotfolioChat.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    UIButton *btn = [UIButton buttonWithType:UIButtonTypeSystem];
    [btn setFrame:CGRectMake(0, 90, self.view.frame.size.width, 40)];
    [btn setTitle:@"Just a button" forState:UIControlStateNormal];
    [btn addTarget:self action:@selector(btnPressed) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:btn];
    
    [BotfolioChat getUnreadCountWithCompletion:^(NSInteger count) {
        [btn setTitle: [NSString stringWithFormat:@"Unread Count: %i", count] forState:UIControlStateNormal];
    }];
    
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(newMessage:) name:[BotfolioChat botfolioChatNewMessageNotification] object:nil];
}

- (void)btnPressed {
    BotfolioChatViewController *vc = [[BotfolioChatViewController alloc] init];
    [self.navigationController pushViewController:vc animated:true];
}
    
- (void)newMessage: (NSNotification*) notification {
    
    Message *m = [notification.userInfo valueForKey:@"message"];
    
    MessageTypes messageType = [BotfolioChat getMessageTypeByInt: m.messageTypeInt];
    
    switch (messageType) {
        case MessageTypesText:
        NSLog(@"%@", m.text);
        break;
        case MessageTypesContact:
        NSLog(@"%@", m.contactName);
        default:
        break;
    }
}
    
    
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: BotfolioChatExampleSwift23/BotfolioChatExampleSwift23/ViewController.swift
//
//  ViewController.swift
//  BotfolioChatExampleSwift23
//
//  Created by Aleks Mutlu on 15/12/2016.
//  Copyright © 2016 Lena Software. All rights reserved.
//

import UIKit
import BotfolioChat

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(newMessage(_:)), name: NSNotificationName.Botfolio.NewMessage, object: nil)
        
        let btn = UIButton(type: .System)
        btn.frame = CGRectMake(0, 90, self.view.frame.width, 40)
        btn.setTitle("Just a button", forState: .Normal)
        btn.addTarget(self, action: #selector(btnPressed), forControlEvents: .TouchUpInside)
        self.view.addSubview(btn)
        
        BotfolioChat.getUnreadCount { (count) in
            btn.setTitle("Unread Count: \(count)", forState: .Normal)
        }
    }
    
    func newMessage(notification: NSNotification) {
        if let message = notification.userInfo?["message"] as? Message {
            if message.messageType == .text {
                print(message.text!)
            }
            else if message.messageType == .contact {
                print(message.contactName)
            }
            
            print("Sent by \(message.sentByBot == true ? "Bot" : "User")")
        }
    }

    func btnPressed() {
        let vc = BotfolioChatViewController()
        self.navigationController?.pushViewController(vc, animated: true)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

